src/Worker.Extensions.DurableTask/DurableHttpResponse.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+/// <summary>
+/// Response received from the HTTP request made by the Durable Function.
+/// </summary>
+public class DurableHttpResponse
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DurableHttpResponse"/> class.
+    /// </summary>
+    /// <param name="statusCode">HTTP Status code returned from the HTTP call.</param>
+    /// <param name="headers">Headers returned from the HTTP call.</param>
+    /// <param name="content">Content returned from the HTTP call.</param>
+    public DurableHttpResponse(
+        HttpStatusCode statusCode,
+        IDictionary<string, StringValues> headers = null,
+        string content = null)
+    {
+        this.StatusCode = statusCode;
+        this.Headers = HttpHeadersConverter.CreateCopy(headers);
+        this.Content = content;
+    }
+
+    /// <summary>
+    /// Status code returned from an HTTP request.
+    /// </summary>
+    [JsonPropertyName("statusCode")]
+    public HttpStatusCode StatusCode { get; }
+
+    /// <summary>
+    /// Headers in the response from an HTTP request.
+    /// </summary>
+    [JsonPropertyName("headers")]
+    [JsonConverter(typeof(HttpHeadersConverter))]
+    public IDictionary<string, StringValues> Headers { get; }
+
+    /// <summary>
+    /// Content returned from an HTTP request.
+    /// </summary>
+    [JsonPropertyName("content")]
+    public string Content { get; }
+
+    /// <summary>
+    /// Creates a DurableHttpResponse from an HttpResponseMessage.
+    /// </summary>
+    /// <param name="httpResponseMessage">HttpResponseMessage returned from the HTTP call.</param>
+    /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+    public static async Task<DurableHttpResponse> CreateDurableHttpResponseWithHttpResponseMessage(HttpResponseMessage httpResponseMessage)
```

Comment:
nit: I do one of two naming methods for these static creation methods: `From{Type*}` (where Type* isn't exact, but a good enough name of what the type is). Or just a `Create`. I tend to prefer the former.

```suggestion
    public static async Task<DurableHttpResponse> FromHttpResponse(HttpResponseMessage httpResponseMessage)
```